File: src/common/py/core/logging/logger_protocol.py
from typing import Protocol


class LoggerProtocol(Protocol):
    """
    Defines the general interface for our extended ``Logger``.
    """
    def debug(self, msg: str, *, scope: str | None = None, **kwargs) -> None: ...
    def info(self, msg: str, *, scope: str | None = None, **kwargs) -> None: ...
    def warning(self, msg: str, *, scope: str | None = None, **kwargs) -> None: ...
    def error(self, msg: str, *, scope: str | None = None, **kwargs) -> None: ...

File: src/common/web/core/logging/LoggerProtocol.ts
/**
 * Defines the general interface for our extended ``Logger``.
 */
export interface LoggerProtocol {
    debug(msg: string, scope: string, params: Record<string, any>): void;

    info(msg: string, scope: string, params: Record<string, any>): void;

    warning(msg: string, scope: string, params: Record<string, any>): void;

    error(msg: string, scope: string, params: Record<string, any>): void;
}
